pidx_viewer.cpp

```diff
@@ -152,9 +161,11 @@ int main(int argc, char **argv) {
     std::vector<char> imgBuf(sizeof(uint32_t) * app.fbSize.x, app.fbSize.y, 0);
     tjDecompress2(decompressor, jpgBuf.data(), imgBytes, imgBuf.data(),
         app.fbSize.x, width * 4, app.fbSize.y, TJPF_RGBA, 0);
-        */
+    */
     std::vector<char> imgBuf(sizeof(uint32_t) * app.fbSize.x * app.fbSize.y, 0);
     ospcommon::read(renderServer, imgBuf.data(), imgBuf.size());
+    imgBufData = (uint32_t*) imgBuf.data();
```

Comment:
This pointer is going to be left dangling, I think just making the `imgBuf` vector a global will work fine for saving the image out. Optionally we could just keep the `jpgBuf` as a global and dump the jpg to disk once we start sending jpgs over the network instead.

Reply:
Hmm I think the `fbSize` can be changed from frame to frame, so it is not a good idea to keep `imgBuf` global. Maybe the best way is setting a boolean flag indicating if the image is ready. When we try to save the image, we have to check the flag first.

Reply:
Right, so each frame we would update the global vector with the new image, overwriting the old one. So it would resize the vector as well if the framebuffer resized

Reply:
Maybe just simply make `imgBuf` global, it will be much easier to read and maintain :D

Reply:
Yea, I think we have some miscommunication, that's exactly what I mean 😆